x-pack/platform/plugins/shared/ml/server/saved_objects/saved_objects.ts

```diff
@@ -27,13 +30,36 @@ interface MlModuleAttributes {
   datafeeds: object[];
 }
 
+const AD_PREFIX = `${AD_SAVED_OBJECT_TYPE}-`;
+const DFA_PREFIX = `${DFA_SAVED_OBJECT_TYPE}-`;
+
 export function setupSavedObjects(savedObjects: SavedObjectsServiceSetup) {
   savedObjects.registerType({
     name: ML_JOB_SAVED_OBJECT_TYPE,
     hidden: false,
     namespaceType: 'multiple',
     migrations,
     mappings: mlJob,
+    management: {
+      importableAndExportable: false,
+      getTitle(obj) {
+        return obj.id;
+      },
+      getInAppUrl(obj) {
+        if (obj.id.startsWith(AD_PREFIX)) {
+          // @TODO: Temporarily hide link until new management link is ready
+          // const jobId = obj.id.slice(AD_PREFIX.length);
+        }
+        if (obj.id.startsWith(DFA_PREFIX)) {
+          // @TODO: Temporarily hide link until new management link is ready
+        }
+        return undefined;
+      },
+      icon: 'machineLearningApp',
+      displayName: i18n.translate('xpack.ml.savedObjects.mlJob.displayName', {
+        defaultMessage: 'Machine Learning',
```

Comment:
Can this say `Machine learning model` for the two job type saved objects, and `Trained model` for the model ones?

<img width="447" alt="Screenshot 2025-01-03 at 17 12 07" src="https://github.com/user-attachments/assets/b1591549-866a-47c8-9177-daf730c6b06d" />

Reply:
Sorry, I should have said `Machine learning job` for the two job type saved objects, and Trained model for the model ones.

Reply:
Text changes in https://github.com/elastic/kibana/pull/205177/commits/602c73f16ea75fb7093b97710596cf47fc26c1c9 look good!